ui/lesson/ScriptingChallenge/Editor.tsx

```diff
@@ -21,22 +21,33 @@ export default function Editor({
   onValidate,
   code,
   constraints,
+  hiddenRange,
 }: {
   language: string
   value?: string
   onChange?: (value: string) => void
   onValidate?: (value: monaco.editor.IMarker[]) => void
   code?: string
   constraints: any
+  hiddenRange?: number[]
 }) {
   const { activeView } = useLessonContext()
   const isActive = activeView === LessonView.Code
   const [loading, setLoading] = useState<boolean>(true)
+  const [options, setOptions] = useState(monacoOptions)
   const monacoRef = useRef<any>()
 
+  const createMonacoOptions = (range) => {
+    return {
+      ...monacoOptions,
+      lineNumbers: (lineNumber: number) => (lineNumber - range).toString(),
```

Comment:
Here is where the  magic happens @benalleng ?

Reply:
yep!